runtime/thunk/thunk.ts

```diff
@@ -124,6 +125,47 @@ function doFindAndExecutePackFunction<T extends FormulaSpecification>({
             return parentFormula.execute(params as any, executionContext);
           }
           break;
+        case MetadataFormulaType.CellAutocomplete:
+          const syncTable = syncTables?.find(table => table.name === formulaSpec.syncTableName);
+          const autocompleteFn = ensureExists(syncTable?.autocompleteCell);
+          const propertyValues = {};
+
+          const cacheKeysUsed: string[] = [];
+
+          function recordPropertyAccess(key: string) {
+            if (!cacheKeysUsed.includes(key)) {
+              cacheKeysUsed.push(key);
+            }
+          }
+
+          for (const [key, value] of Object.entries(formulaSpec.propertyValues)) {
+            Object.defineProperty(propertyValues, key, {
+              get() {
+                recordPropertyAccess(key);
+                return value;
+              },
+            });
+          }
+
+          const cellAutocompleteExecutionContext: Omit<CellAutocompleteExecutionContext, 'search'> = {
+            ...executionContext,
+            propertyName: formulaSpec.propertyName,
+            propertyValues,
+          };
+
+          Object.defineProperty(cellAutocompleteExecutionContext, 'search', {
+            get() {
+              recordPropertyAccess('__search');
+              return formulaSpec.search;
+            },
+          });
+
+          const result = await autocompleteFn.execute(params as any, cellAutocompleteExecutionContext);
+          return {
+            result,
+            propertiesUsed: cacheKeysUsed,
+          } as any;
```

Comment:
We'll want to type this.

Reply:
Added some types for the return value. We may need to iterate on it in the future, but as a first crack it looks like this:

```
/**
 * These will be created with a helper function.
 */
interface PropertyAutocompleteFormattedResult {
  __brand: 'PropertyAutocompleteSimpleResult';
  /** Text that will be displayed to the user in UI for this option. */
  display: string;
  /** The actual value for this option */
  value: any;
}

type PropertyAutocompleteResults =
  | any[]
  | {
      cacheTtlSecs?: number;
      results: Array<any | PropertyAutocompleteFormattedResult>;
    };

/**
 * @hidden
 */
export interface PropertyAutocompleteAnnotatedResult {
  packResult: PropertyAutocompleteResults;
  propertiesUsed: string[];
}
```

This doesn't address the question of how to distinguish `PropertyAutocompleteFormattedResult` from a pack object that also has fields named "display" and "value". Seems like distinguish those cases we'd need a helper function either to wrap individual results (and fill in the `__brand` value, maybe?) or to wrap the entire result list.